src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -190,29 +223,17 @@ end
 # and jth reference coordinate, respectively. These are geometric terms which
 # appear when using the chain rule to compute physical derivatives as a linear
 # combination of reference derivatives.
-@inline function get_contravariant_vector(element, orientation, mesh::DGMultiMesh{1})
-  @unpack rxJ = mesh.md
-  return 2 * SVector(rxJ[1, element]) # the 1D contravariant vector reduces to a scaling.
+@inline function get_contravariant_vector(element, orientation, mesh::DGMultiMesh{NDIMS}) where {NDIMS}
+  # note that rstxyzJ = [rxJ, sxJ, txJ; ryJ syJ tyJ; rzJ szJ tzJ], so that this will return
+  # 2 * SVector{2}(rxJ[1, element], ryJ[1, element]) in 2D.
+  @unpack rstxyzJ = mesh.md
+  return 2 * SVector{NDIMS}(getindex.(rstxyzJ[:, orientation], 1, element))
 end
 
-@inline function get_contravariant_vector(element, orientation, mesh::DGMultiMesh{2})
-  @unpack rxJ, sxJ, ryJ, syJ = mesh.md
-  if orientation == 1
-    return 2 * SVector(rxJ[1, element], ryJ[1, element])
-  else # if orientation == 2
-    return 2 * SVector(sxJ[1, element], syJ[1, element])
-  end
-end
-
-@inline function get_contravariant_vector(element, orientation, mesh::DGMultiMesh{3})
-  @unpack rxJ, sxJ, txJ, ryJ, syJ, tyJ, rzJ, szJ, tzJ = mesh.md
-  if orientation == 1
-    return 2 * SVector(rxJ[1, element], ryJ[1, element], rzJ[1, element])
-  elseif orientation == 2
-    return 2 * SVector(sxJ[1, element], syJ[1, element], szJ[1, element])
-  else # if orientation == 3
-    return 2 * SVector(txJ[1, element], tyJ[1, element], tzJ[1, element])
-  end
+@inline function get_contravariant_vector(element, orientation, mesh::DGMultiMesh{NDIMS, NonAffine()}) where {NDIMS}
+  # note that rstxyzJ = [rxJ, sxJ, txJ; ryJ syJ tyJ; rzJ szJ tzJ]
+  @unpack rstxyzJ = mesh.md
+  return 2 * SVector{NDIMS}(view.(rstxyzJ[:, orientation], :, element))
```

Comment:
Same here

Reply:
This does allocate, but I think it's because of the `2 *` in front. `rstxzyJ` is a `SMatrix` so `getindex` and broadcasting seem to be fast. 

I'll try to move the `2 *` around to avoid this.